servers/v3-proxy-api/src/graph/toGraphQL.spec.ts

```diff
@@ -3,25 +3,125 @@ import { setSaveInputsFromGetCall } from './toGraphQL';
 import {
   SavedItemsSortBy,
   SavedItemsSortOrder,
-  SavedItemStatusFilter,
 } from '../generated/graphql/types';
+import { defaultQuery } from '../test/fixtures';
 
 describe('toGraphQL', () => {
-  //todo: refactor this test as you refactor the method
   it('should map saves input', () => {
     const expected = {
       pagination: {
         limit: 30,
         offset: 0,
       },
       sort: {
-        sortBy: SavedItemsSortBy.UpdatedAt,
-        sortOrder: SavedItemsSortOrder.Asc,
-      },
-      filter: {
-        status: SavedItemStatusFilter.Unread,
+        sortBy: SavedItemsSortBy.CreatedAt,
+        sortOrder: SavedItemsSortOrder.Desc,
       },
+      filter: {},
     };
-    expect(setSaveInputsFromGetCall('sample-rest-params')).toEqual(expected);
+    expect(setSaveInputsFromGetCall(defaultQuery)).toEqual(expected);
+  });
+  describe('filter object', () => {
+    // Need to see how Express handles this
+    it.todo('handles duplicate query parameters');
+    it.each([
+      {
+        params: { since: 123456 },
+        expected: { filter: { updatedSince: 123456 } },
+      },
+      {
+        params: { state: 'all' as const },
+        expected: { filter: {} },
+      },
+      {
+        params: { favorite: true, tag: 'recipe' },
+        expected: { filter: { isFavorite: true, tagNames: ['recipe'] } },
+      },
+      {
+        params: {
+          state: 'read' as const,
+          unused: 'hello',
+          contentType: 'article' as const,
+        },
+        expected: {
+          filter: {
+            status: 'ARCHIVED',
+            contentType: 'IS_READABLE',
+          },
+        },
+      },
+    ])(
+      'maps filters appropriately, including combos',
+      ({ params, expected }) => {
+        const actual = setSaveInputsFromGetCall({ ...defaultQuery, ...params });
+        console.log(JSON.stringify(actual, null, 2));
+        expect(actual).toMatchObject(expected);
+      },
+    );
+  });
+  describe('sort', () => {
+    it.each([
+      {
+        params: { since: 123456 },
+        expected: { sort: { sortBy: 'CREATED_AT', sortOrder: 'DESC' } },
+      },
+      {
+        params: {
+          favorite: true,
+          tag: 'recipe',
+          sort: 'newest' as const,
+        },
+        expected: { sort: { sortBy: 'FAVORITED_AT', sortOrder: 'DESC' } },
```

Comment:
Took me a min to realize why we sort as Favorited At 🤣  had to think about the UI there.